utils/wasm-gen/src/generator/syscalls/imports.rs

```diff
@@ -280,22 +280,21 @@ impl<'a, 'b> SysCallsImportsGenerator<'a, 'b> {
                 .into();
             initial_mem_size.memory_size()
         };
+
         // subtract to be sure we are in memory boundaries.
-        let reserve_gas_result_ptr = memory_size_in_bytes.saturating_sub(100) as i32;
-        let rid_pid_value_ptr = reserve_gas_result_ptr + mem::size_of::<Length>() as i32;
-        let pid_value_ptr = reserve_gas_result_ptr + mem::size_of::<ErrorWithHash>() as i32;
-        let reservation_send_result_ptr = pid_value_ptr + mem::size_of::<HashWithValue>() as i32;
+        let pid_value_ptr = memory_size_in_bytes.saturating_sub(100) as i32;
+        let rid_pid_value_ptr = pid_value_ptr + mem::size_of::<HashWithValue>() as i32;
```

Comment:
`HashWithValue` is allocated here, but `gr_reservation_send` takes `rid_pid_value: *const TwoHashesWithValue`
looks like need to fix the instructions since `*mut ErrorWithHash` comes from arguments and we can no longer construct `TwoHashesWithValue` in easy way